src/oscar/templatetags/category_tags.py

```diff
@@ -1,12 +1,33 @@
+from six import with_metaclass
 from django import template
-
 from oscar.core.loading import get_model
 
 register = template.Library()
 Category = get_model("catalogue", "category")
 
 
-class CheapCategoryInfo(dict):
+def create_passthrough(name):
+    def category_passthrough(self):
+        return getattr(self.category, name)
+    return category_passthrough
+
```

Comment:
This is the same as the `__getattr__` implementation, but less readable, and we'll still have the multiple function calls.
You'll want something like https://github.com/Pylons/pyramid/blob/master/src/pyramid/decorator.py#L4 , or keep the `__getattr__` implementation as it is probably fast enough.

Reply:
I know this information at parse time, so no need to modify __dict__ at runtime.
Also this is definately not the same as the __getattr__.

Reply:
For every field on the model, a passthrough methid is added that directs the call to self.category. so these are in ``__dict__`` at parse time and not added at runtime, which is very ugly.